impress-org/givewp
# Feature: hide gateways when donation amount is zero

Resolves [GIVE-2118]

## Description


This hides gateways when the donation amount is zero.  This is primarily a precautionary measure to avoid errors with gateways as there is no reason to display a gateway when there is no amount. It's purely a visual update to prevent gateways from rendering in this scenario.  It does not actually prevent the form from submitting, which will let the validation framework take over.

Related: https://github.com/impress-org/givewp/pull/7515

In the future, our forms may be more flexible in creating something that does not equate to an amount.  However, at this time all our gateways require an amount.


## Affects



The v3 donation form gateway options

## Visuals



<img width="693" alt="Screenshot 2025-01-17 at 1 38 44 PM" src="https://github.com/user-attachments/assets/6f71a47a-5525-4dc4-b3bb-2e764cf115e9" />


## Testing Instructions



Enter $0 in the custom amount field and scroll to the gateway options, you should be presented with a message instead of gateways. 

Zip: https://github.com/impress-org/givewp/actions/runs/12912508739

## Pre-review Checklist






[GIVE-2118]: https://stellarwp.atlassian.net/browse/GIVE-2118?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
